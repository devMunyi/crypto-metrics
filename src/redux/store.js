import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import cryptoReducer from './reducers';

export default configureStore(
  {
    reducer: {
      crypto: cryptoReducer,
    },
  },
  applyMiddleware(logger),
);
