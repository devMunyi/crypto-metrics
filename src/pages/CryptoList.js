import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { SyncOutlined } from '@ant-design/icons';
import { BsFillArrowRightCircleFill } from 'react-icons/bs';
import { NavLink } from 'react-router-dom';
import { fetchCryptos } from '../redux/actions';
import Headline from '../components/Headline';
import truncateNum from '../helpers';

function CryptoList() {
  // search values
  const [searchInput, setInputSearch] = useState('');

  // set dispatch hook
  const dispatch = useDispatch();

  // execute fetch crypto list on page mount
  useEffect(() => {
    dispatch(fetchCryptos());
  }, [dispatch]);

  // get the cryptos from store
  const { cryptoList, isLoading } = useSelector((state) => state.crypto);

  const cryptoCount = cryptoList.length;

  // const handleSearchFilter = () => {
  //   cryptoList = cryptoList.filter(({ name }) => name === searchInput);
  // };

  const handleSearchChange = (e) => {
    setInputSearch(e.target.value.trim());

    console.log(searchInput);
  };

  return (
    <div className="container">
      {isLoading && (
        <SyncOutlined
          spin
          style={{ fontSize: '32px' }}
          className="d-flex justify-content-center align-items-center"
        />
      )}
      {!isLoading && (
        <>
          <Headline page="list" cryptoCount={cryptoCount} cryptoDetails={{}} />

          <div className="row pt-4 pb-4 container fw-normal">
            <div className="col">
              {' '}
              <input
                onChange={handleSearchChange}
                name="search"
                className="form-control"
                type="search"
                placeholder="Type name or price value..."
              />
            </div>
            <div className="col">
              <button type="button" className="btn btn-outline-light">
                Search
              </button>
            </div>
          </div>

          <div className="grid-tmp">
            {cryptoList.map((crypto) => (
              <div key={crypto.id} className="grid-tmp-item p-2 text-end">
                <div className="grid-item-top">
                  <NavLink
                    style={{ color: '#fff' }}
                    to={`/details/${crypto.id}`}
                  >
                    <BsFillArrowRightCircleFill
                      title="Click to view Details"
                      role="presentation"
                      className="hover"
                    />
                  </NavLink>
                </div>

                <div className="grid-item-bottom fw-normal">
                  <h4>{crypto.name.toUpperCase()}</h4>
                  <p>
                    {truncateNum(crypto.priceUsd, 8)}
                    {' '}
                    usd
                  </p>
                </div>
              </div>
            ))}
          </div>
        </>
      )}
    </div>
  );
}

export default CryptoList;
