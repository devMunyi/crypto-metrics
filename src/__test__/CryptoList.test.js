import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import CryptoList from '../pages/CryptoList';

it('renders CryptoList Component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <CryptoList />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
