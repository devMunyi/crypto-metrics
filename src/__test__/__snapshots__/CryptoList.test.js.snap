// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CryptoList Component correctly 1`] = `
<div
  className="container"
>
  <div
    className="row border headline"
    style={
      Object {
        "boxShadow": "rgba(0, 0, 0, 0.24) 0px 3px 8px",
      }
    }
  >
    <div
      className="col"
      style={
        Object {
          "height": "200px",
        }
      }
    >
      <img
        alt="CRYPTO"
        className="img-fluid"
        src="cryptos.png"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="col d-flex justify-content-center align-items-center border-start"
    >
      <div
        className="div"
      >
        <h4>
          CRYPTO METRICS
        </h4>
        <p
          className="fw-normal"
        >
           
          0 cryptocurrencies
        </p>
      </div>
    </div>
  </div>
  <div
    className="row pt-4 pb-4 container fw-normal"
  >
    <div
      className="col"
    >
       
      <input
        className="form-control"
        name="search"
        onChange={[Function]}
        placeholder="Type name or price value..."
        type="search"
      />
    </div>
    <div
      className="col"
    >
      <button
        className="btn btn-outline-light"
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  <div
    className="grid-tmp"
  />
</div>
`;
